src/mjolnir/ingest_transit.cc

```diff
@@ -438,12 +476,34 @@ bool write_stop_pair(Transit& tile,
       service_dow->Add(trip_calendar->friday == gtfs::CalendarAvailability::Available);
       service_dow->Add(trip_calendar->saturday == gtfs::CalendarAvailability::Available);
       service_dow->Add(trip_calendar->sunday == gtfs::CalendarAvailability::Available);
-      // skip if no days are valid
-      if (!trip_calendar || !service_dow->size()) {
-        LOG_WARN("Service ID " + currTrip->service_id + " has no calendar entry, skip.");
+
+      const gtfs::CalendarDates& trip_calDates = feed.get_calendar_dates(currTrip->service_id);
+      bool had_added_date = false;
+      for (const auto& cal_date_item : trip_calDates) {
+        auto d = to_local_epoch_sec(cal_date_item.date.get_raw_date());
+        if (cal_date_item.exception_type == gtfs::CalendarDateException::Added) {
+          stop_pair->add_service_added_dates(d);
+          had_added_date = true;
+        } else
+          stop_pair->add_service_except_dates(d);
+      }
+
+      // this shouldn't happen, but let's make sure it doesn't
+      // in convert_transit we'll check if there was a valid date for this service and skip if not
+      if (!service_dow->size() && !had_added_date) {
+        LOG_WARN("Service ID " + currTrip->service_id +
+                 " has no valid calendar or calendar_dates entry, skipping...");
+        tile.mutable_stop_pairs()->RemoveLast();
         continue;
       }
 
+      // test this, but careful, we might have to adjust the test's dist_shape_traveled or whatever
+      // for the test shapes to be a bit more realistic with the actual map where it travels much
+      // further than the GTFS objects indicate
+      if (!currShape.empty()) {
+        stop_pair->set_shape_id(pbf_shape_it->second);
+      }
```

Comment:
only set a shape_id when it's not empty. the shape is being used by convert_transit only if there's a `shape_id` set